pkg/kubeapiserver/options/authentication.go

```diff
@@ -888,3 +870,12 @@ func loadAuthenticationConfigFromData(data []byte) (*apiserver.AuthenticationCon
 
 	return configuration, nil
 }
+
+// areFlagsConfigured is a function that returns true if any of the oidc-* flags are configured.
+func (o *OIDCAuthenticationOptions) areFlagsConfigured() bool {
+	return !reflect.DeepEqual(o, NewBuiltInAuthenticationOptions().WithOIDC())
```

Comment:
this isn't equivalent, this wouldn't catch someone explicitly setting a flag to the default

Reply:
fs.Changed returns true if Set was called, even if it was set to the same value

Reply:
Hm, right. Reverted the changes done around `areFlagsSet`

Reply:
There are also enough repackaging of kube-apiserver like k3s, microshift etc. It's not guaranteed that the values actually come from real flags.

Reply:
> Hm, right. Reverted the changes done around `areFlagsSet`

the change now looks ~fine, does it actually resolve the oidc integration test initialization issue you were seeing?

Reply:
Yes the test is passing with these changes + the flagz changes.